### PR TITLE
Spare freshness checks

### PR DESCRIPTION
**Pull request by tilacog**
This reorders when and if freshness checks are performed in a polling cycle.

Currently, the Oracle performs a freshness check at every iteration, even though we are not expecting a new epoch.

If merged, this will only execute a freshness check if the subgraph is either uninitialized or in a previous epoch than the Epoch Manager.

### CODE REVIEW

File: crates/oracle/src/contracts.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ where
             .query("currentEpoch", (), None, Default::default(), None)
             .await?;
         let current_epoch = epoch_number.as_u64();
-        debug!(current_epoch, "Got current epoch from the Epoch Manager");
+        debug!("Epoch Manager is at epoch {current_epoch}");
```
Comment (tilacog) [inline crates/oracle/src/contracts.rs:52]: This is just for using the same log format as the [Subgraph epoch check](https://github.com/graphprotocol/block-oracle/pull/160/files#diff-e8e985bdb5ff7e8247748de894aaebb17322ce87813dadce3a8f269dcf4dcf7bR146).

File: crates/oracle/src/oracle.rs [hunk 2]
```diff
@@ -63,49 +63,53 @@ impl Oracle {
     /// if necessary.
     pub async fn run(&mut self) -> Result<(), Error> {
         info!("New polling iteration.");
-        if !self.detect_new_epoch().await? {
+        if self.detect_new_epoch().await? {
+            self.handle_new_epoch().await?;
+        } else {
             debug!("No epoch change detected.");
-            return Ok(());
         }
```
Comment (neysofu) [inline crates/oracle/src/oracle.rs:70]: Much nicer!

**Comment by neysofu** [comment]
Reviewing.

**Review by neysofu — APPROVED** [review]

**Comment by neysofu** [comment]
There's a small conflict.

**Pull request merged**